awscli/customizations/paginate.py

```diff
@@ -225,6 +234,13 @@ def check_should_enable_pagination(input_tokens, shadowed_args, argument_table,
             argument_table[key] = value
 
 
+def check_and_warn_negative_max_items(parsed_args, parsed_globals, **kwargs):
+    parsed_max_items = getattr(parsed_args, 'max_items', None)
+    if parsed_globals.paginate and parsed_max_items is not None:
+        if int(parsed_max_items) <= 0:
+            logger.critical("Non-positive values for --max-items are unsupported and may yield undefined behavior.")
```

Comment:
For warnings in the AWS CLI, we have a couple conventions:
* Use `uni_print` to print to stderr, like [this example](https://github.com/aws/aws-cli/blob/v2/awscli/customizations/s3/subcommands.py#L1315)
* We format the string to start with `warning: `

Reply:
Thanks, I addressed this in my latest revision.